pkg/controller/service/service.go

```diff
@@ -13,25 +16,58 @@ func RenderServices(req router.Request, resp router.Response) error {
 	// reset, should get repopulated
 	svcInstance.Status.Endpoints = nil
 
-	objs, _, err := services.ToK8sService(req, svcInstance)
-	if err != nil {
-		return err
-	}
-	resp.Objects(objs...)
-	svcInstance.Status.HasService = len(objs) > 0
+	var svcList []*v1.ServiceInstance
+	http2Ports := ports.ByProtocol(svcInstance.Spec.Ports, true, v1.ProtocolHTTP2)
+	httpPorts := ports.ByProtocol(svcInstance.Spec.Ports, true, v1.ProtocolHTTP)
+	// If we have both types of ports we need to create a separate service with the annotation
+	if len(http2Ports) > 0 && len(httpPorts) > 0 {
```

Comment:
I'll put some explanation on this, since we paired up to work on this yesterday and I still feel like this is a good and simple solution that can be a little confusing.

The background is that we have to put a traefik annotation on the K8s Service for http2 support (h2c). With that service we'd break the traffic for "normal" http traefik to other ports listed in that service though.
E.g. if someone uses http2 for the app server, but metrics are served via http.
That means, if we have both http/http2 mixed in a single ServiceInstance, we have to split it up to generate two K8s services.

However, all the code further down the stream is specific to a single service and even re-uses the ServiceInstance name in a lot of places.
So we need a "virtual" ServiceInstance with a different name, holding only the http2 ports to be able to leave lots of code untouched.

The DeepCopies are necessary to leave the original serviceInstance object untouched during that process and we only have to populated actively modified status fields on it.